scripts/zones/Hazhalm_Testing_Grounds/mobs/Hakenmann.lua

```diff
@@ -0,0 +1,61 @@
+-----------------------------------
+-- Area: Hazhalm Testing Grounds
+--   NM: Hakenmann (Einherjar)
+-- Notes: Uses the same TP move 3 times in a row.
+-----------------------------------
+mixins =
+{
+    require('scripts/mixins/draw_in'),
+}
+-----------------------------------
+---@type TMobEntity
+local entity = {}
+
+local function notBusy(mob)
+    local action = mob:getCurrentAction()
+    if
+        action == xi.act.MOBABILITY_START or
+        action == xi.act.MOBABILITY_USING or
+        action == xi.act.MOBABILITY_FINISH
+    then
+        return false
+    end
+
+    return true
+end
+
+entity.onMobInitialize = function(mob)
+    xi.einherjar.onBossInitialize(mob)
+end
+
+entity.onMobSpawn = function(mob)
+    mob:setLocalVar('tpMoveCount', 3)
+end
+
+entity.onMobWeaponSkillPrepare = function(mob, target)
+    if mob:getLocalVar('tpMoveCount') == 0 then
+        mob:setLocalVar('tpMoveCount', 3)
+    end
+
+    return mob:getLocalVar('tpMove')
+end
+
+entity.onMobWeaponSkill = function(target, mob, skill)
+    -- TODO: If one move is stunned, the sequence is stopped
```

Comment:
Do we have a binding or listener for is a mob is stunned? Seems like a good idea